app/models/invenio_rdm_record_converter.rb

```diff
@@ -246,14 +244,17 @@ def build_creator_contributor_json(creator)
           "given_name": given_name,
           "family_name": family_name,
           "type": "personal",
-          "identifiers": identifiers
         }
       }
+
+      if dh_user.orcid.present?
+        identifiers = [{"scheme": "orcid", "identifier": dh_user.orcid.split('/').pop}]
+        json.merge!({"identifiers": identifiers})
+      end
     # Personal record without user in database
     elsif creator.include?(",")
       json = {
-        "person_or_org":
-        {
+        "person_or_org": {
           "given_name": given_name,
           "family_name": family_name,
           "type": "personal"
```

Comment:
What is the practical difference between this branch and the one above? They seem to result in same output. If nothing else was lost in the merge, then this can be simplified further:

```ruby
if organization?(creator)
...
else
  dh_user = User.find_by(formal_name: creator) || User.find_by(display_name: display_name)
  if dh_user.present?
    ...
    if dh_user.orcid.present?
    ...
    end
  else
    json = {
        "person_or_org": {
          "name": creator,
          "type": "organizational"
        }
      }
end
```

This should also reduce the DB/Solr  calls.

Reply:
I think that previously there was significant differences in what we wanted from a creatibutor in the database versus a creatibutor that was not in the database, but yeah with this change we can just remove the branch